examples/solvers/denoising_with_entropy_type_regularization

```diff
@@ -0,0 +1,148 @@
+step sizes tau and sigma such that tau sigma ||K||_2^2 < 1, and set the
+total number of iterations.
+
+For details see :ref:`chambolle_pock`, :ref:`proximal_operators`, and
+references therein.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+import scipy
+import scipy.ndimage
+import matplotlib.pyplot as plt
+
+import odl
+
+
+# TODO: Use BroadCastOperator instead of ProductSpaceOperator
+
+# Read test image:
+# convert integer values to float, and rotate to get the image upright
+image = np.rot90(scipy.misc.ascent()[::2, ::2], 3).astype('float')
+shape = image.shape
+
+# Rescale
+image *= 100 / image.max()
+
+# Discretized spaces
+discr_space = odl.uniform_discr([0, 0], shape, shape)
+
+# Original image
+orig = discr_space.element(image)
+
+# Add noise
+image = np.random.poisson(1 + image)
+
+# Data of noisy image
+noisy = discr_space.element(image)
+
+# Gradient operator
+gradient = odl.Gradient(discr_space, method='forward')
+
+# Matrix of operators
+op = odl.ProductSpaceOperator([[odl.IdentityOperator(discr_space)],
+                               [gradient]])
+
+# Starting point
+x = op.domain.zero()
+
+# Operator norm
+prod_op_norm = 1.1 * odl.operator.oputils.power_method_opnorm(op, 100)
+print('Norm of the product space operator: {}'.format(prod_op_norm))
+
+# Proximal operator related to the primal variable
+
+# Non-negativity constraint
+proximal_primal = odl.solvers.proximal_nonnegativity(op.domain)
+
+# Proximal operators related to the dual variable
+
+# l2-data matching
+prox_convconj_kl = odl.solvers.proximal_convexconjugate_kl(discr_space,
+                                                           lam=1, g=noisy)
+
+# TV-regularization: l1-semi norm of grad(x)
+prox_convconj_l1 = odl.solvers.proximal_convexconjugate_l1(gradient.range,
+                                                           lam=1/5)
+
+# Combine proximal operators: the order must match the order of operators in K
+proximal_dual = odl.solvers.combine_proximals([prox_convconj_kl,
+                                               prox_convconj_l1])
+
+# Optional: pass partial objects to solver
+partial = (odl.solvers.PrintIterationPartial() &
+           odl.solvers.util.PrintTimingPartial() &
+           odl.solvers.util.ShowPartial())
+
+# Number of iterations
+niter = 100
+
+# Step size for the proximal operator for the primal variable x
+tau = 1 / prod_op_norm
+
+# Step size for the proximal operator for the dual variable y
+sigma = 1 / prod_op_norm
+
+# Run algorithms (and display intermediates)
+odl.solvers.chambolle_pock_solver(
+    op, x, tau=tau, sigma=sigma, proximal_primal=proximal_primal,
+    proximal_dual=proximal_dual, niter=niter, partial=partial)
+
+# Display images
+orig.show(title='original image')
+noisy.show(title='noisy image')
+x.show(title='reconstruction')
```

Comment:
by doing show=True here you can skip plt.show() in the next row and thus avoid the import up top. This allows us to change backend later on without pain here.